src/main.rs

```diff
@@ -1,61 +1,73 @@
 #![allow(unused)]
-mod libs;
 mod arguments;
+mod libs;
 
 // use std::env::args;
 // use std::path::PathBuf;
-use termion::{clear, color, style, terminal_size, raw::IntoRawMode, cursor, input::TermRead, event::Key};
 use core::time;
-use std::{io::{self, Read, Write}, thread, time::{Instant, Duration}};
-use libs::matrix::Matrix;
 use libs::drawable::Drawable;
+use libs::matrix::Matrix;
+use std::{
+    io::{self, Read, Write},
+    thread,
+    time::{Duration, Instant},
+};
+use termion::{
+    clear, color, cursor, event::Key, input::TermRead, raw::IntoRawMode, screen, style,
+    terminal_size,
+};
 
 use arguments::parse_cli_arguments;
 
 fn main() {
     // read cli arguments
     let settings = parse_cli_arguments();
-    
 
     // lock stdout
     let stdout = io::stdout();
     let mut stdout = stdout.lock(); // stdout handle
     let mut stdout = stdout.into_raw_mode().unwrap(); // 정교한 제어를 위해 raw mode
-    
+
     // lock stdin
     let mut stdin = termion::async_stdin();
     let mut it = stdin.keys();
-    
+
     let terminal_size = termion::terminal_size().ok();
-    let width = terminal_size.map(|(w,_)| w).unwrap_or(100);
-    let height = terminal_size.map(|(_,h)| h).unwrap_or(100);
+    let width = terminal_size.map(|(w, _)| w).unwrap_or(100);
+    let height = terminal_size.map(|(_, h)| h).unwrap_or(100);
 
     // set buffer size
     let mut stdout = io::BufWriter::with_capacity(width as usize * height as usize * 2, stdout);
-        
+
     // initialize matrix
     let mut matrix = Matrix::new(width, height, &settings);
 
     // start drawing
     let frames_per_second: u16 = settings.frames;
-    let mili_per_frame: u16 = 1000/frames_per_second;
+    let mili_per_frame: u16 = 1000 / frames_per_second;
     let mut frame_count: u16 = 0;
-    write!(stdout, "{}{}", cursor::Hide, clear::All);
+    write!(
+        stdout,
+        "{}{}{}",
+        screen::ToAlternateScreen,
+        cursor::Hide,
+        clear::All
```

Comment:
It would be nice if you could comment on where the actual change is

Reply:
So, as far as I understand, whole thing will write on another screen
and when the loop ends, it will return to the restored terminal right?